src/util/prolog/prolog.test.ts

```diff
@@ -0,0 +1,296 @@
+
+import { PrologSession } from "./prologSession";
+import * as program from "./testPrograms";
+
+// Expected results are consistent with SWI-Prolog behavior except that they usually have an extra "false."
+
+describe("Tau-Prolog Behavior, PrologSession.executeQuery() & PrologResult.getResultArray()", () => {
```

Comment:
Why is there one `describe` for three different things? Usually if you want to nest `describe` blocks it should be something like 
```
describe("my super nice module", () => {
  describe("some function", () => {
    it("does great things", () => {...});
  });
});
```
Arbitrarily grouping things together doesn't add much value. I've also never needed more than one level of `describe` - usually there aren't that many different functions to test in one file, so it's sufficient to use one `describe` block for each function, and have one `it` / `test` for each test case.